spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaAutoConfiguration.java

```diff
@@ -138,4 +145,38 @@ public KafkaAdmin kafkaAdmin() {
 		return kafkaAdmin;
 	}
 
+	@Configuration
+	@ConditionalOnClass(StreamsBuilder.class)
+	public static class KafkaStreamsConfiguration {
+
+		@Bean(KafkaStreamsDefaultConfiguration.DEFAULT_STREAMS_CONFIG_BEAN_NAME)
+		public Properties defaultKafkaStreamsConfig(KafkaProperties properties) {
```

Comment:
I am not keen to expose a `@Bean` of type `Properties`, perhaps we could have a dedicated object for this or does the underlying configuration actually require this?

Reply:
Kafka deprecated the use of an immutable `StreamsConfig` in favor of a `Properties` object to configure the `StreamsBuilder`. I can add a wrapper class in spring-kafka if you prefer. 

https://jira.apache.org/jira/browse/KAFKA-6386

Reply:
Yes please, `Properties` is way to generic to be exposed as a bean.